### PR TITLE
th-112: Add query parameters service

### PR DESCRIPTION
**Pull request by h0wter**
All functions work with either single or multiple arguments.
`getQueryParameters('param')` or `getQueryParameters('param', 'param1')`
`removeQueryParameters('param')` or `removeQueryParameters('param', 'param1')`
`setQueryParameters({param: value})` or `setQueryParameters({param: value, param1: value1})`
`setQueryParameters` can take both a string and a number as a property value.

### CODE REVIEW

File: frontend/src/libs/hooks/use-query-parameters/use-query-parameters.hook.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { useCallback, useSearchParameters } from '../hooks.js';
```
Comment (PavloSukhinin) [inline frontend/src/libs/hooks/use-query-parameters/use-query-parameters.hook.ts]: Circular dependency, be careful with that

**Review by Druxight — DISMISSED** [review]

File: frontend/src/libs/hooks/use-query-parameters/use-query-parameters.hook.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+type Queries = Record<string, string | null>;
+
+type Parameters = Record<string, string | number>;
```
Comment (PavloSukhinin) [inline frontend/src/libs/hooks/use-query-parameters/use-query-parameters.hook.ts]: rename this, Parameters is build-in TS type

**Review by PavloSukhinin — CHANGES_REQUESTED** [review]

**Review by DmytroVoronkov — APPROVED** [review]

File: frontend/src/libs/hooks/use-query-parameters/use-query-parameters.hook.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      setSearchParameters(newSearchParameters);
+    },
+    [searchParameters, setSearchParameters],
+  );
+
+  const removeQueryParameters = useCallback(
+    (...keys: string[]): void => {
+      const newSearchParameters = new URLSearchParams(searchParameters);
+
+      if (keys.length > 1) {
```
Comment (Some14u) [inline frontend/src/libs/hooks/use-query-parameters/use-query-parameters.hook.ts:48]: consider when there is no parameters being passed. I think you can add "remove everything" for that particular case.

**Review by Some14u — APPROVED** [review]
Simple and clean. I like it, however in my opinion it lacks possibility to pass and handle default values.
For example, if i need some key value, which has it's own default, i'd expect this to return the default value when there is no corresponding key in the query.

**Review by PavloSukhinin — APPROVED** [review]

**Review by kalinkaaaa14 — APPROVED** [review]

**Pull request merged**